Frameworks/UIKit/UIImage.mm

```diff
@@ -714,102 +710,108 @@ - (void)drawAsPatternInRect:(CGRect)pos {
-                                        img->_imageInsets.left,
-                                        getImage(img)->Backing()->Height() - img->_imageInsets.top - img->_imageInsets.bottom),
-                               makeRect(0,
-                                        img->_imageInsets.bottom,
-                                        img->_imageInsets.left,
-                                        ii->size.size.height - img->_imageInsets.top - img->_imageInsets.bottom));
-        if (getImage(img)->Backing()->Width() - img->_imageInsets.left - img->_imageInsets.right > 0) {
+        if (src->size.width - leftCap - rightCap > 0) {
+            // MidHeightMidWidth
             CGContextDrawImageRect(context,
-                                   getImage(img),
-                                   makeRect(img->_imageInsets.left,
-                                            img->_imageInsets.bottom,
-                                            getImage(img)->Backing()->Width() - img->_imageInsets.left - img->_imageInsets.right,
-                                            getImage(img)->Backing()->Height() - img->_imageInsets.top - img->_imageInsets.bottom),
-                                   makeRect(img->_imageInsets.left,
-                                            img->_imageInsets.bottom,
-                                            ii->size.size.width - img->_imageInsets.left - img->_imageInsets.right,
-                                            ii->size.size.height - img->_imageInsets.top - img->_imageInsets.bottom));
+                                   cgImg,
+                                   makeRect(leftCap, botCap, (src->size.width - leftCap - rightCap), (src->size.height - topCap - botCap)),
+                                   makeRect((dst->origin.x + leftCap),
+                                            (dst->origin.y + botCap),
+                                            (dst->size.width - leftCap - rightCap),
+                                            (dst->size.height - topCap - botCap)));
         } else {
-            assert(0);
+            UNIMPLEMENTED_WITH_MSG("Patched draws only supported when sum of leftCap and rightCap is less than the width of the UI element.");
+        }
+
+        if (rightCap) {
+            // MidHeightRight
+            CGContextDrawImageRect(context,
+                                   cgImg,
+                                   makeRect((src->size.width - rightCap), botCap, rightCap, (src->size.height - topCap - botCap)),
+                                   makeRect((dst->origin.x + dst->size.width - rightCap),
+                                            (dst->origin.y + botCap),
+                                            rightCap,
+                                            (dst->size.height - topCap - botCap)));
+        }
+    } else {
+        UNIMPLEMENTED_WITH_MSG("Patched draws only supported when sum of topCap and botCap is less than the height of the UI element.");
```

Comment:
That's true and this is a pre-existing issue. I will probably create another pull request in the future to address this and potentially a functional test for this scenario to make sure it matches the output on the reference platform.